### DISCUSSION TITLE
System freezes when stream doesn't have focus and disable external requests on web front-end? privacy concern?

### DISCUSSION THREAD

**Discussion by accessiblepixel** [LizardByte/community#405] [Sunshine] [answered]
I don't know where to post this and I've thought about various ways about asking about this, although since it's not something I know how to solve or what other stuff is required it's made me kind of give up reporting it every time until I pushed through this time.

I have Sunshine installed on a Windows 10 22H2 system, and I connect from another Windows 10 22H2 system in a window - if I'm doing something that takes an amount of time like extracting a large number of archives or installing a GOG game, if I don't have the client focused all these apps freeze until I actively view the stream again, an I'm not sure what causes it. Once I tab to the app again, things resume as if nothing had happened, it's like it's entered some kind of standby or sleep maybe, like today when doing something in WinRAR it stopped and it's estimated completion just rose to abotu 499 hours until I tabbed back in after setting it going and then it resumed.

My other concern is that the web client has external dependencies and what seems to be user tracking enabled by default? How do I opt out of these and have it so that the web client (and system client) runs without loading any external resources by default? except maybe an opt in update check?

These are the domains I found during my testing that look like they're collecting data from my webconfig session... 

static.cloudflareinsights.com
e.widgetbot.io
cdn.ravenjs.com
stonks.widgetbot.io
cdn.discordapp.com
app.lizardbyte.dev

**Comment by ReenigneArcher** [comment, marked as answer]
> static.cloudflareinsights.com
e.widgetbot.io
cdn.ravenjs.com
stonks.widgetbot.io
cdn.discordapp.com
app.lizardbyte.dev

They're all for the discord integration. It's a cloud service so it has to be online.

On your other point, I have no idea. Never heard of anyone else reporting anything like that. I don't think it has anything to do with Sunshine. Maybe your host is going into some kind of energy saving mode?

**Reply by accessiblepixel** [reply]
It isn't going into powersave though, since it's all still displayed on the actual machine's display.

I kinda knew reporting it would be difficult, hence it's taken me a long time to even want to jump in to try and figure it out...  If I'm doing the same things through remote desktop though - no issue.

I had a check through logs to see if it was hanging up / reporting anything but yeah it just hangs there if the window doesn't have focus on some tasks - usually disk related like installers or un-archiving.

--- privacy mitigation

Since you said it was some kind of discord integration, I figured a workaround for blocking those 3rd party requests, a ublock rule for stops it loading... The update check uses github to check for the latest version, so that shouldn't(TM) break. I'll see if I can add it to network DNS so I don't have to configure it per machine.

Ublock Rule
```
* app.lizardbyte.dev * block
```

Edit:

If you can tell me where specifically might give me a start on where to solve it since I'm out of ideas... for the past few months I've just been trying to always have it on top if i'm doing something that takes time... but of course sometimes I can forget and then realise a task that I thought'd be complete hours ago hasn't been running and a thorough search of issues, docs and discussion has not found anyone else with the same issue, so figured I had to report it in case anyone else had the same, and especially like me didn't feel all that confident about reporting it...


Edit 2:

Hmm... your powersave had me questioning things, even though the console's screen had remained on Windows had seemingly reset my 'Turn display off after: 15 minutes' maybe that's a setting that sneaked it's way back to being turned on and causing this? Since when I set my powerprofile to max performance I set it not to turn things off... Weird.

Edit 3:

Okay. Reset that to not turn off. I've got a long extraction task to run... let's see if it makes any difference.

**Reply by accessiblepixel** [reply]
Yeah, it seemed that setting (put display to sleep) sneaking it's way back on was causing issues, even though the system shouldn't have turned off the display, I guess since display turns off my nvidia GPU doesn't like streaming while the screen is asleep and it kind of suspends any graphical tasks (like installers or unarchivers) - command line stuff still runs fine.

Thanks for the pointer @ReenigneArcher think I've fixed it although do feel a bit like an idiot since I struggled with it for months and it turned out (so far) to be something simple.

I've blocked app.lizardbyte.dev on my network at the DNS level which stops it loading discord.js and all the other domains I'd rather not load for the webadmin panel, at least until there's an opt out for that or something. 

Much appreciated.

**Reply by ReenigneArcher** [reply]
There's not going to be an opt out. The discord integration is to provide an easier way for users to get support, see announcements, etc.
